src/info/infra/repositories/DataverseInfoRepository.ts

```diff
@@ -0,0 +1,27 @@
+import { IDataverseInfoRepository } from '../../domain/repositories/IDataverseInfoRepository';
+import axios, { AxiosResponse } from 'axios';
+import { ReadError } from '../../../core/domain/repositories/ReadError';
+import { DataverseVersion } from '../../domain/models/DataverseVersion';
+
+export class DataverseInfoRepository implements IDataverseInfoRepository {
+  private apiUrl: string;
+
+  constructor(apiUrl: string) {
+    this.apiUrl = apiUrl;
+  }
+
+  public async getDataverseVersion(): Promise<DataverseVersion> {
+    const response = await axios.get(`${this.apiUrl}/info/version`).catch((error) => {
+      throw new ReadError(error.response.status + error.response.data ? ': ' + error.response.data.message : '');
+    });
```

Comment:
```
const response = await axios.get(`${this.apiUrl}/info/version`)
                      .then((response) => getVersionFromResponse(response))
                      .catch((error) => throw new ReadError(error.message));

```

I really prefer being able to read the whole path sequentially. then.then.then.catch

Reply:
I agree, more visual!